### ISSUE TITLE
row names in phenotype data

### ISSUE THREAD

**Issue by Nairobi-2020** [bhklab/MetaGxBreast#9]
after loading TCGA data, I found that row.names(phenotype) is different from the columns "sample_name" "alt_sample_name".
the row.names(phenotype) is identical with colnames(expr).
I tend to believe that the programmer initially got error message since row.names(phenotype) != colnames(expr), and then forced them to be equal by assigning:
row.names(phenotype) = colnames(expr)

Then, this is only to avoid error message when creating eset object, and row.names(phenotype) is incorrect and we should ignore.

An I correct?

**Comment by Nairobi-2020** [comment]
I found that 5 datasets in the package has the sample name problems.
[1] "DUKE:    FALSE   FALSE   TRUE"
[1] "EXPO:    FALSE   FALSE   TRUE"
[1] "STNO2:    FALSE   FALSE   TRUE"
[1] "TCGA:    FALSE   FALSE   TRUE"
[1] "TRANSBIG:    FALSE   FALSE   TRUE"

The 3 columns with logic are:
  a = identical(row.names(pheno), pheno$sample_name)
  b = identical(names(mrna), pheno$sample_name)
  c = identical(names(mrna), row.names(pheno))

**Comment by Nairobi-2020** [comment]
with some character substitutions, I can match the sample names for other datasets, except TCGA.
I think there is a strong contradiction in the data, and the question is which one I should stick with?

**Comment by Nairobi-2020** [comment]
This is getting a bit, quite a bit frustrating.

So I downloaded TCGA data from cBioportal, and compared the clinical data from there and from MetaGxBreast.
They cannot match up. I tried both way, trusting sample name from expr data and trusting sample name from the sample name column in phenotype data.

This is frustrating because I think I need to discard all computations so far.

**Comment by DarioS** [comment]
It is because the column names of the expression matrix must be valid R syntax. See `?make.names` for more information.

```r
> make.names("DUKE_T01-145")
  "DUKE_T01.145"
```

R applies `make.names` to all column names of expression table, so the developer had to also do it to the clinical row names.

TCGA has different versions of clinical data. The values depend on the snapshot date. cBioPortal provides outdated data. The latest data is provided by [Genomic Data Commons](https://portal.gdc.cancer.gov/) but cBioPortal has Broad Firehose legacy version.

**Comment by Nairobi-2020** [comment]
No, the data there is wrong, you messed up with sample names on different
occasions.
besides, the data is totally out of date, so many much powerful data came
out
